src/ucx_plugin.c

```diff
@@ -595,12 +560,43 @@ ncclResult_t nccl_ucx_regmr_dmabuf(void* comm, void* data, size_t size, int type
 	return nccl_ucx_regmr(comm, data, size, type, mhandle);
 }
 
-ncclResult_t ucx_send_check(ucx_send_comm_t *comm) {
-  ucp_tag_message_h     msg_tag;
-  ucp_tag_recv_info_t   info_tag;
-  ucx_request_t         *req;
-  connect_msg_t         *msg;
-  ucp_ep_params_t       ep_params;
+static ucx_request_t *ucx_request_get(ucx_comm_t *comm) {
+  static const size_t entries = sizeof(comm->reqs) / sizeof(*comm->reqs);
+  ucx_request_t *req;
+
+  for (int i = 0; i < entries; i++) {
+    req = &comm->reqs[i];
+    if (req->used == 0) {
+      req->worker  = comm->worker;
+      req->pending = 0;
+      req->count   = 0;
+      req->used    = 1;
+      return req;
```

Comment:
seems too slow to iterate over the array like that
can we use a freelist? or bitmap (FFS) if num of requests is <=64 ?

Reply:
added freelist